### ISSUE TITLE
Incremental Selection maps can be active only after init_selection

### ISSUE THREAD

**Issue by sencer** [nvim-treesitter/nvim-treesitter#509]
Using key mappings can become difficult as they get lengthier. Sometimes this is not really avoidable, but sometimes it is. I think it is avoidable in case of incremental selection.

node_incremental, node_decremental and scope_incremental mappings can be activated only after init_selection, so one can use simple mappings {"n", "N", "s"} without hurting the original use of these maps.

**Comment by kyazdani42** [comment]
indeed, that's what i do in my configs. We'll take a look at that once we are done with refactoring out these modules.

**Comment by bfredl** [comment]
As it happens, I actually want the reverse: use the same key for `init_selection` and then `node_incremental`. But it doesn't seem to work though to map them to the same physical key/map string?

**Comment by quantum-booty** [comment]
@bfredl  You could bind them on the same key. You could use leader key instead of ctrl if you want to.
```
incremental_selection = {
    enable = true,
    keymaps = {
      init_selection = "<C-n>",
      node_incremental = "<C-n>",
      scope_incremental = "<C-s>",
      node_decremental = "<C-r>",
    },
  },

```

**Comment by bfredl** [comment]
It didn't work before but I will try latest version tomorrow.

**Comment by gregd** [comment]
@quantum-booty Binding `init_selection` and `node_incremental` to the same key is what most people want. Maybe the main docs should have that as an example. It took me a while to find this comment thread.

**Comment by vigoux** [comment]
Hi thanks for reviving this issue and indeed, having the same mapping for init selection and grow selection will be better, I'll make that change.

**Comment by haolian9** [comment]
I was trying to combine incrmental_selection with [vim-submode](https://github.com/kana/vim-submode) to sove this issue, **failed**, althrough, I think it's worth to mention here.

```
call submode#enter_with('inc_sel', 'n', '', 'gnn', ':lua require"nvim-treesitter.incremental_selection".init_selection()<cr>')
call submode#map('inc_sel', 'x', '', 'n', ':lua require"nvim-treesitter.incremental_selection".node_incremental()<cr>')
call submode#map('inc_sel', 'x', '', 'N', ':lua require"nvim-treesitter.incremental_selection".node_decremental()<cr>')
call submode#map('inc_sel', 'x', '', 's', ':lua require"nvim-treesitter.incremental_selection".scope_incremental()<cr>')
```
